components/javascript-wrapper.js

```diff
@@ -0,0 +1,111 @@
+
+/**
+ * This file contacts the common code used by all of the variations of any RDF
+ * node componentin the noflo implementation of the RDF Pipeline.
+ */
+
+var _ = require('underscore');
+var stateComponent = require('./state-component');
+
+/**
+ * This module provides a Javascript wrapper for updaters.  It is responsible for setting up the
+ * code to ensure updater callback, and then invoking the RDF pipeline framework to complete building
+ * the component and setting up the vni and state metadata.
+ *
+ * @param def updater definition. This can be a simple updater function, or an object that contains 
+ *        updater metadata (e.g., input/output port definitions, description, icons), and an updater.
+ *
+ * @return a promise to create the noflo rdf component
+ */
+module.exports = function(def) { 
+
+    var self = this;
+
+    if ( _.isUndefined(def) || ( _.isEmpty(def) && ! _.isFunction( def ))) { 
+        // Nothing to work with - no metadata or updater - can't proceed.
+        throw new Error('javascript-wrapper found no updater definition');
```

Comment:
Let's provide a default input port and a defaultJavascriptUpdater if nothing was specified.  That will make it easy for people to create a node as a stub and then modify it later.  The defaultJavascriptUpdater would just pass the input to the output.  BTW, by default an input port should not allow multiple connections.